tools/doc-generator/doc-generator.go

```diff
@@ -56,66 +96,34 @@ func parseVirtMetrics(r io.Reader, contains []string) map[string]string {
 	return metrics
 }
 
-func writeToFile(metrics map[string]string) {
-	old, err := os.Open("metrics.md")
+func writeConstMarkdown(markdown string) {
+	doc, err := os.OpenFile(docFileName, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0755)
 	checkError(err)
-	defer old.Close()
-
-	new, err := os.Create("newmetrics.md")
-	checkError(err)
-	defer new.Close()
-
-	var write string
-
-	format := "###%s"
+	defer doc.Close()
 
-	buf := bufio.NewWriter(new)
-	scan := bufio.NewScanner(old)
-	for scan.Scan() {
-		line := scan.Text()
-		if write != "" {
-			if strings.Contains(line, "HELP") {
-				if line != write {
-					buf.WriteString(write)
-					line = ""
-				}
-
-			} else {
-				buf.WriteString(write)
-				buf.WriteString("\n")
-			}
-			write = ""
-
-		}
-
-		buf.WriteString(line + "\n")
-		for k, v := range metrics {
-			if strings.Contains(line, k) {
-				write = fmt.Sprintf(format, v)
-				delete(metrics, k)
-				break
-			}
-		}
-	}
-	if scan.Err() != nil {
-		panic(fmt.Errorf("Failed to update metric doc, %v", scan.Err()))
-	}
+	doc.WriteString(markdown)
+	doc.WriteString("\n")
+}
 
-	if len(metrics) > 0 {
-		buf.WriteString("\n # Other Metrics \n")
-	}
+// writeMetricsMarkdown can be used as a generic writer for metrics
+func writeMetricsMarkdown(metrics map[string]string, substring string) {
+	var markdown string
 
-	var keys []string
+	keys := make([]string, len(metrics))
 	for k := range metrics {
 		keys = append(keys, k)
 	}
-
+	// Necessary to always write metrics in same order
 	sort.Strings(keys)
-	for _, k := range keys {
-		buf.WriteString("## " + k + "\n")
-		buf.WriteString("###" + metrics[k] + "\n")
+
+	for _, metric := range keys {
+		if strings.Contains(metric, substring) {
+			markdown += "#### " + metric + "\n"
+			markdown += metrics[metric] + "\n"
+			markdown += "\n\n"
+		}
 	}
-	buf.Flush()
```

Comment:
What is the reason for the removal of the flush?

Reply:
To be honest, I still couldn't understand what it does. Might have deleted unintentionally and it didn't make any apparent difference.

Should I re-add it?

Reply:
Honestly I didn't look close enough and so I missed that you changed the implementation completely away from using a buffer, which I think is a good improvement.

The only thing I don't like here that the file is opened and closed in method `writeConstMarkdown`, which is called several times. Normally I'd rather object against such usage, as opening and closing a file is an I/O operation, which like every operation that takes more costly resources should be reduced to the minimum. To put it short, I'd prefer opening the `doc` in method `main` and passing it through to the methods as a variable.

But this is just a general remark, regarding the limited number of executions I'd be ok with it. I'd leave it to you to change it according to my suggestion.